client/tasks/fills/purchase.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { registerPlugin } from '@wordpress/plugins';
+import { WooOnboardingTaskListItem } from '@woocommerce/onboarding';
+import { useState, useCallback } from '@wordpress/element';
+import { recordEvent } from '@woocommerce/tracks';
+import { useSelect } from '@wordpress/data';
+import { ONBOARDING_STORE_NAME, PLUGINS_STORE_NAME } from '@woocommerce/data';
+
+/**
+ * Internal dependencies
+ */
+import CartModal from '../../dashboard/components/cart-modal';
+import { getCategorizedOnboardingProducts } from '../../dashboard/utils';
+
+const PurchaseTaskItem = () => {
+	const [ cartModalOpen, setCartModalOpen ] = useState( false );
+
+	const { installedPlugins, profileItems } = useSelect( ( select ) => {
+		const { getProfileItems } = select( ONBOARDING_STORE_NAME );
+		const { getInstalledPlugins } = select( PLUGINS_STORE_NAME );
+
+		return {
+			installedPlugins: getInstalledPlugins(),
+			profileItems: getProfileItems(),
+		};
+	} );
+
+	const toggleCartModal = useCallback( () => {
+		if ( ! cartModalOpen ) {
+			recordEvent( 'tasklist_purchase_extensions' );
+		}
+
+		setCartModalOpen( ! cartModalOpen );
+	}, [ cartModalOpen ] );
+
+	const groupedProducts = getCategorizedOnboardingProducts(
+		profileItems,
+		installedPlugins
+	);
+	const { remainingProducts } = groupedProducts;
+
+	return (
+		<WooOnboardingTaskListItem id="purchase">
+			{ ( { defaultTaskItem, onClickActions } ) => (
+				<>
+					{ React.cloneElement( defaultTaskItem, {
+						onClick: () => {
+							onClickActions();
```

Comment:
Is there any way we can handle these onclick events in the parent?  I'm concerned consumers may forget to implement and we lose tracking.

Reply:
Yeah, something that I'd been considered considering as well. I'll look into this real quick.

Reply:
I've reworked this in 64841e097b6d439067ac7fa595addc1d55e668a6.